packages/loader/container-loader/src/prefetchDocumentStorageService.ts

```diff
@@ -31,6 +32,12 @@ export class PrefetchDocumentStorageService extends DocumentStorageServiceProxy
         return this.cachedRead(blobId);
     }
 
+    public async readBlob(blobId: string): Promise<ArrayBufferLike> {
+        const res: Promise<ArrayBufferLike> = this.cachedRead(blobId).then((value) => {
```

Comment:
Code would be more readable if it was like that:

value = await this.cachedRead(blobId);
return stringToBuffer(value, "base64");

That said, it is incorrect it is not very efficient, as it forces (in worse case) conversion of binary -> base64 -> binary.
Better would be for cachedRead to not return base64-encoded data, but return binary, by calling readBlob, not read().
And then callers to figure out how to adapt result to their needs (i.e. this.read() to do conversion from binary to base64).

Reply:
I didn't want to change cachedRead or read in this PR. just adding readBlob and tests

Reply:
But you will be making it less efficient the moment code starts to use readBlob instead of read(). So this PR, or next, we need to make it as fast as it was before.